k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -399,13 +429,39 @@ StatusCode EDM4hep2LcioTool::convertCollections(lcio::LCEventImpl* lcio_event) {
   }
   debug() << "Event: " << lcio_event->getEventNumber() << " Run: " << lcio_event->getRunNumber() << endmsg;
 
-  // Deal with
   EDM4hep2LCIOConv::sortParticleIDs(pidCollections);
+  if (!m_podioDataSvc) {
+    DataObject* p;
+    StatusCode  code = m_eventDataSvc->retrieveObject("/Event" + k4FWCore::frameLocation, p);
+    if (code.isSuccess()) {
+      auto* frame = dynamic_cast<AnyDataWrapper<podio::Frame>*>(p);
+      edmEvent    = std::cref(frame->getData());
+    } else {
+      auto frame = podio::Frame{};
+      edmEvent   = frame;
```

Comment:
This will lead to a dangling reference, right? The `frame` that `edmEvent` is referring to will be destroyed at the end of the `else` scope and I don't think `reference_wrapper` does any lifetime extension?

Reply:
I don't think so, `std::reference_wrapper` is basically a pointer to the object so only the pointer gets deleted: https://en.cppreference.com/w/cpp/utility/functional/reference_wrapper. Also if that was the case, since this function is called many times and there is only one frame then it would crash when the function is called a second time.